geode-core/src/main/java/org/apache/geode/internal/cache/BackupManager.java

```diff
@@ -569,6 +570,28 @@ private void createBackupDir(File backupDir) throws IOException {
     }
   }
 
+  private void backupOplog(File targetDir, Oplog oplog) throws IOException {
+    File crfFile = oplog.getCrfFile();
+    backupFile(targetDir, crfFile);
+
+    File drfFile = oplog.getDrfFile();
+    backupFile(targetDir, drfFile);
+
+    oplog.finishKrf();
+    File krfFile = oplog.getKrfFile();
+    backupFile(targetDir, krfFile);
+  }
+
+  private void backupFile(File targetDir, File file) throws IOException {
+    if (file != null && file.exists())
+      try {
+        Files.createLink(targetDir.toPath().resolve(file.getName()), file.toPath());
```

Comment:
Is there any reason you cannot use a symbolic link here? That way it would allow the link to span filesystems too.

Reply:
The use of a hard link here is to replace the existing logic of making a copy of the original file for the backup. Deletion of the original file cannot be allowed to result in loss of that file in the backup location and thus a symbolic link cannot be used in this case. Hard links can be used here in place of copying the file only because the Oplog files being copied are guaranteed to be read-only.

Reply:
OK - that makes sense. Thanks for the explanation.